mm2src/coins/lp_coins.rs

```diff
@@ -1395,6 +1430,96 @@ impl WithdrawError {
     }
 }
 
+#[derive(Serialize, Display, Debug, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum SignatureError {
+    #[display(fmt = "Invalid request: {}", _0)]
+    InvalidRequest(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    InternalError(String),
+    #[display(fmt = "Coin is not found: {}", _0)]
+    CoinIsNotFound(String),
+    #[display(fmt = "Message prefix not found")]
+    PrefixNotFound,
+}
+
+impl HttpStatusCode for SignatureError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            SignatureError::InvalidRequest(_) => StatusCode::BAD_REQUEST,
+            SignatureError::CoinIsNotFound(_) => StatusCode::BAD_REQUEST,
+            SignatureError::InternalError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            SignatureError::PrefixNotFound => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl From<keys::Error> for SignatureError {
+    fn from(e: keys::Error) -> Self { SignatureError::InternalError(e.to_string()) }
+}
+
+impl From<ethkey::Error> for SignatureError {
+    fn from(e: ethkey::Error) -> Self { SignatureError::InternalError(e.to_string()) }
+}
+
+impl From<PrivKeyNotAllowed> for SignatureError {
+    fn from(e: PrivKeyNotAllowed) -> Self { SignatureError::InternalError(e.to_string()) }
+}
+
+impl From<CoinFindError> for SignatureError {
+    fn from(e: CoinFindError) -> Self { SignatureError::CoinIsNotFound(e.to_string()) }
+}
+
+#[derive(Serialize, Display, Debug, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum VerificationError {
+    #[display(fmt = "Invalid request: {}", _0)]
+    InvalidRequest(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    InternalError(String),
+    #[display(fmt = "Signature decoding error: {}", _0)]
+    SignatureDecodingError(String),
+    #[display(fmt = "Address decoding error: {}", _0)]
+    AddressDecodingError(String),
+    #[display(fmt = "Coin is not found: {}", _0)]
+    CoinIsNotFound(String),
+    #[display(fmt = "Message prefix not found")]
```

Comment:
Could you please change it to something like sign_message_prefix is not set in coin config?